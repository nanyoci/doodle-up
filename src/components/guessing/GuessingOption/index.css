.option{
    font-family: 'Merriweather Sans', sans-serif;
    font-weight: bold;
    font-size: 1.2vw;
    width: 70%;
}

.option-unit{
    padding-top: 30px;
}

.audio-icon{
    max-height: 100%;
    margin-right: 10px;
    cursor: pointer;
}