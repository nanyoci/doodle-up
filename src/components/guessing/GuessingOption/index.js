import React, { useState } from 'react';
import useSound from 'use-sound';

import audioIcon from '../../../assets/audio.svg';
import './index.css';

export default function GuessingOption(props) {
	const {
		text,
		audio,
		isComplete,
		isCorrect,
		onSubmit,
	} = props;

	const [isClicked, setIsClicked] = useState(false);
	const [playAudio] = useSound(audio);
	const handleSubmit = () => {
		setIsClicked(true);
		onSubmit(isCorrect);
	}

	let className = "option btn";

	if (!isClicked)
		className += " btn-outline-dark";
	else {
		if (isCorrect)
			className += " btn-primary";
		else
			className += " btn-danger";
	}

	return (
		<div className="option-unit">
			<img src={audioIcon} className="audio-icon" alt="audio" onClick={playAudio} />
			<button
				className={className}
				disabled={isComplete || isClicked}
				onClick={handleSubmit}
			>
				{text}
			</button>
		</div>
	)
}
