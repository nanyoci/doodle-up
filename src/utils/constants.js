export const API_URL = 'https://doodle-up-backend.herokuapp.com';