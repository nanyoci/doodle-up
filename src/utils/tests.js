import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}