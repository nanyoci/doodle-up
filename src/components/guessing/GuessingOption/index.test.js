import React from 'react'
import { cleanup, act } from '@testing-library/react'

import GuessingOption from './index.js';
import { renderWithReduxRouter } from './../../../utils/tests';

jest.mock('axios');
console.warn = jest.fn()

beforeEach(() => {
    window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.addTextTrack = () => { /* do nothing */ };
})

afterEach(() => {
    cleanup();
    jest.clearAllMocks();
});


it('should take a snapshot', async () => {
    let asFragment = {};
    await act(async () => { ({ asFragment } = renderWithReduxRouter(<GuessingOption />)) });
    expect(asFragment()).toMatchSnapshot();
})
