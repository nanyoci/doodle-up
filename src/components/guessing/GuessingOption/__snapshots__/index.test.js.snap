// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="option-unit"
  >
    <button
      alt="audio"
      class="ui violet large circular icon button"
    >
      <i
        aria-hidden="true"
        class="volume up icon"
      />
    </button>
    <button
      class="option btn btn-outline-dark"
    />
  </div>
</DocumentFragment>
`;
